### DISCUSSION TITLE
Can I rely on `group$.key` from `groupBy`?

### DISCUSSION THREAD

**Discussion by Sawtaytoes** [ReactiveX/rxjs#6101] [Q&A] [answered]
I need an easy way to access the key from `groupBy`. From what I saw, I can do `group$.key`. Dunno if I should be relying on `.key` being there.

Here's what I have:
```js
groupBy(({
  groupKey,
}) => (
  groupKey
)),
mergeMap((
  group$,
) => (
  group$
  .pipe(
    pluck('data'),
    toArray(),
    map((
      dataArray,
    ) => ({
      dataArray,
      key: (
        group$
        .key
      ),
    })),
  )
)),
```

Without adding a `mergeMap`, I have no way of getting the passed-in key after `toArray`. That's why I thought to use `group$.key`. Is this how someone would recommend I do it?

**Comment by cartant** [comment, marked as answer]
Yes, it's an exported type:

https://github.com/ReactiveX/rxjs/blob/8fdc00d3b3a06c76648814ad2e628c5828e7b789/src/internal/operators/groupBy.ts#L262-L271

**Reply by Sawtaytoes** [reply]
Thanks! That's awesome.

I must've missed it when I was looking the docs yesterday. I see it now though 🤷‍♂️:

![image](https://user-images.githubusercontent.com/3948069/110383730-52b88580-8022-11eb-9028-e77a51625ea2.png)
